tests/regression/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls::{
+    config::Builder,
+    security,
+    testing::{CertKeyPair, InsecureAcceptAllCertificatesHandler},
+};
+type Error = s2n_tls::error::Error;
+
+// Function to create default config with specified parameters
+pub fn set_config(
+    cipher_prefs: &security::Policy,
+    keypair: CertKeyPair,
+) -> Result<s2n_tls::config::Config, Error> {
+    let mut builder = Builder::new();
+    builder
+        .set_security_policy(cipher_prefs)
+        .expect("Unable to set config cipher preferences");
+    builder
+        .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
+        .expect("Unable to set a host verify callback.");
+    builder
+        .load_pem(keypair.cert(), keypair.key())
+        .expect("Unable to load cert/pem");
+    builder.trust_pem(keypair.cert()).expect("load cert pem");
+    builder.build()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crabgrind as cg;
+    use s2n_tls::testing::TestPair;
+    use std::{
+        env,
+        fs::{create_dir, File},
+        io::{self, BufRead, Write},
+        path::Path,
+        process::Command,
+    };
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
```

Comment:
```suggestion
    /// configurable threshold for regression testing. Test cases will fail if ...
    const COST: u64 = 1_000_000;
```
nit one is that there is nice IDE integrations with "doc comments" (specified by `///` instead of `//`). IDE's will show the doc comment whenever someone mouses-over the constant.

Additionally, let's add a little bit more information about this. I'm assuming that this is the maximum jump in instruction count allowed? I have to make a bit of a logic jump to figure that out from the comment though

Reply:
Agreed, I also want to adjust the const name for more clarity to `MAX_DIFF` so that's included in the revision.